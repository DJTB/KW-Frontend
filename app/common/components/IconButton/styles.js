import styled from 'styled-components';

import { gutter } from 'common/styles/layout';
import { resetButton } from 'common/styles/utils';
import { fastEaseQuad } from 'common/styles/animation';

export const Button = styled.button`
  ${resetButton}
  ${gutter()}
  display: flex;
  justify-content: center;
  align-self: center;
  align-items: center;
  vertical-align: middle;
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;
  transform: scale(1);

  &:not(:disabled) {
    &:focus,
    &:hover {
      opacity: .9;
      outline: none;
    }

    &:active {
      opacity: 1;
      transform: scale(.9);
    }
  }

  &:disabled {
    cursor: not-allowed;
  }
`;
