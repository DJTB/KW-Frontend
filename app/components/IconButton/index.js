import React from "react";
import PropTypes from "prop-types";

import Icon from "components/Icon";
import Button from "base/Button";

// FIXME: just use <Button plainButton><Icon /></Button> instead..... geez

IconButton.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.oneOf(["submit", "reset", "button"]),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  inline: PropTypes.bool,
  children: PropTypes.node,
  plainButton: PropTypes.bool,
};

IconButton.defaultProps = {
  type: "button",
  plainButton: true,
  color: "currentColor",
  size: "1.5em",
  disabled: false,
  inline: false,
  children: null,
  onClick: (event) => event /* passthrough, for submit buttons in forms with onSubmit */,
};

function IconButton({
  name, title, color, size, inline, children, ...props
}) {
  return (
    <Button aria-label={title} {...props}>
      {children}
      <Icon name={name} inline={inline} size={size} color={color} />
    </Button>
  );
}

export default IconButton;
