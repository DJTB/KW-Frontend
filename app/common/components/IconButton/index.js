import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { gutter } from 'common/styles/layout';

import Aux from 'common/components/Aux';
import Button from 'common/components/Button';
import Icon from 'common/components/Icon';

// prettier-ignore
export const Text = styled.span`
  ${gutter({ position: 'left' })}
  ${gutter({ position: 'right', mod: 2 })}
`;

IconButton.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.oneOf(['submit', 'reset', 'button']),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  inline: PropTypes.bool,
  plainButton: PropTypes.bool,
  text: PropTypes.any,
  isSubmitting: PropTypes.bool,
};

IconButton.defaultProps = {
  type: 'button',
  plainButton: true,
  color: 'currentColor',
  size: '1.5em',
  disabled: false,
  inline: false,
  text: '',
  isSubmitting: false,
  onClick: (event) => event /* passthrough, for submit buttons in forms with onSubmit */,
};

function IconButton({ name, text, title, color, size, inline, isSubmitting, ...props }) {
  const RenderedText = text ? <Text>{text}</Text> : null;
  const RenderedIcon = (
    <Icon
      isRotating={isSubmitting}
      name={isSubmitting ? 'SYNC' : name}
      inline={inline}
      size={size}
      color={color}
    />
  );

  return (
    <Button {...props} title={title} aria-label={title} color={color}>
      <Aux>
        {RenderedText}
        {RenderedIcon}
      </Aux>
    </Button>
  );
}

export default IconButton;
