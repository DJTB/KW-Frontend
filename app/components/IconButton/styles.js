import styled from 'styled-components';

import { resetButton } from 'shared/styles/utils';
import { fastEaseQuad } from 'shared/styles/animation';

export const Button = styled.button`
  ${resetButton}
  transition: all ${fastEaseQuad}, transform 100ms linear;
  cursor: pointer;
  opacity: .7;

  &:not(:disabled) {
    &:focus,
    &:hover {
      opacity: .9;
      outline: none;
    }

    &:active {
      opacity: 1;
      transform: scale(.9);
    }
  }

  &:disabled {
    cursor: not-allowed;
  }
`;
