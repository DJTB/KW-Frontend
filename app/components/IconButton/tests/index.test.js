import React from 'react';
import { shallow } from 'enzyme';
import IconButton from '../index';

describe('<IconButton /> ', () => {
  it('should match baseline snapshot of required props', () => {
    const renderedComponent = shallow(
      <IconButton name="ADD" title="Does an action" handleClick={jest.fn()} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
