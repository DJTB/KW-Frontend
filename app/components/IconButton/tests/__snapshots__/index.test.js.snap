// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton />  should adopt button types 1`] = `
<button
  aria-label="Does an action"
  class="jZIabh"
  title="Does an action"
  type="submit"
>
  <span
    class="vwFqP"
    color="currentColor"
    display="block"
  >
    <svg
      class="hfAVjD"
      height="100%"
      preserveaspectratio="xMidYMid meet"
      title="ADD"
      viewbox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</button>
`;

exports[`<IconButton />  should match baseline snapshot 1`] = `
<button
  aria-label="Does an action"
  class="jZIabh"
  title="Does an action"
  type="button"
>
  <span
    class="vwFqP"
    color="currentColor"
    display="block"
  >
    <svg
      class="hfAVjD"
      height="100%"
      preserveaspectratio="xMidYMid meet"
      title="ADD"
      viewbox="0 0 24 24"
      width="100%"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
      />
    </svg>
  </span>
</button>
`;
